# Problem 2: RNA-seq analysis ğŸ§¬
In this repository you will find some scriptsğŸ’» that I used to adress different issues that my group ecountered while solvingğŸ’¡ an exercicie in the context of the subject "**Bacterial Genomics**"ğŸ”¬. 
All of them are **python** scripts that use famous packagesâš™ï¸ such as *matplotlib*, *pandas* and *numpy*.

## ğŸ“ `FPKMs_plotting`
- ğŸ“„ `gene_count_fpkm.txt`: FPKM values for each gene and for each replica of the experiment. It is the input.
- ğŸ’» `FPKM_barplot.py`: script that plots the distributionğŸ“ˆ of FPKMs in all the replicas.
- ğŸ¨ `plt_FPKM.png`: FPKM distribution plotted.

## ğŸ“ `FoldChange_plotting`
- ğŸ“„ `Mutant_vs_WildType_results.txt`: file with normalized FPKMs for each replica, with calculated fold change (FC) and p-value. 
- ğŸ’» `FC_histplot.py`: script that plots the distributionğŸ“ˆ of the fold changes, regardless of their p-value. 
- ğŸ¨ `plt_FC.png`: FC distribution plotted. 
  
## ğŸ“ `COG_terms`
- ğŸ“„ `Locus_tag_filtered_results.txt`: locus tags for each gene that has been defined as differentially expressed in our data (previously filteringâŒ¨ï¸ made). It is one of the inputs.
- ğŸ“„ `Stenotrophomonas_maltophilia_K279a_ASM7248v1_genomic.g2d.COG`: COG (Clusters of Orthologous Genes) terms for each gene of the reference S. maltophilia K279a genome. It is one of the imputs.
- ğŸ’» `COGs.py`: script that uses the COG terms for our DEGs.
- ğŸ“„ `DEGs_with_COGs.csv`: csv output fileğŸ”§ with each DEG and its COG term. 
